python/mxnet/gluon/data/dataloader.py

```diff
@@ -112,6 +113,9 @@ def default_mp_batchify_fn(data):
 
 def worker_loop(dataset, key_queue, data_queue, batchify_fn):
     """Worker loop for multiprocessing DataLoader."""
+    if isinstance(dataset, RecordFileDataset):
```

Comment:
don't do this kind of special casing.

Reply:
how do you suggest to do it then?

Reply:
@piiswrong we could also have a `reload` method on `Dataset` that does nothing for all dataset except for the RecordFileDataset ones, that way there is no special casing.